docs/scripts/update-docs/update-components-docs.py

```diff
@@ -161,11 +189,18 @@ def get_otel_components(version='main', component_docs_mapping=None):
     for comp in otel_components:
         # Extract the component name without the suffix (e.g., 'filelogreceiver' from 'filelogreceiver ')
         comp_name = comp['name'].strip()
+        
+        # Check if this component is deprecated (takes precedence)
+        if comp_name in deprecated_components:
+            comp['support_status'] = 'Deprecated'
+            comp['is_deprecated'] = True
```

Comment:
this is not being used anywhere, yet, right?
i.e. would need to update https://github.com/elastic/elastic-agent/blob/main/docs/scripts/update-docs/templates/components-table.jinja2 as well to mark the deprecated components in the rendered table